
# HG changeset patch
# User John H. Robinson, IV <jaqque@sbih.org>
# Date 1225082572 25200
# Node ID ce1b81cb49375becb2268da560cb586f9aaf7bb6
# Parent e07d85ba8a2719cdd2e50715405bac2cc7e69f6c
Workaround for Composite problems when default depth is not 24.

  This is really a workaround patch, the proper fix would be to have
WindowMaker know about Composite, and take advantage of its features.

Author: Paulo Cesar Pereira de Andrade <pcpa@mandriva.com.br>

--- a/src/main.c	Thu Apr 27 12:16:58 2006 -0700
+++ b/src/main.c	Sun Oct 26 21:42:52 2008 -0700
@@ -828,6 +828,16 @@
         exit(1);
     }
 
+    if (wVisualID < 0)
+	/*
+	 *   If unspecified, use default visual instead of waiting
+	 * for wrlib/context.c:bestContext() that may end up choosing
+	 * the "fake" 24 bits added by the Composite extension.
+	 *   This is required to avoid all sort of corruptions when
+	 * composite is enabled, and at a depth other than 24.
+	 */
+	wVisualID = (int)DefaultVisual(dpy, DefaultScreen(dpy))->visualid;
+
     /* check if the user specified a complete display name (with screen).
      * If so, only manage the specified screen */
     if (DisplayName)

